space_invaders.py

```diff
@@ -1 +1,295 @@
+import sys, pygame, random
+from pygame import K_LEFT, K_RIGHT, K_UP, K_DOWN
 
+pygame.init()
+
+DELAY = 100 # delay between graphic refresh cycles in ms
+ALIEN_WINGS_DOWN_IMAGE_FILE = "alien.bmp"
+ALIEN_WINGS_UP_IMAGE_FILE = "alien_wings_up.png"
+PLAYER_IMAGE_FILE = "cannon.jpg"
+LASER_IMAGE_FILE = "laser.jpg"
+
+SCREEN_ROWS = 10 # number of rows that aliens descend through
+SIZE = WIDTH, HEIGHT = (600, 400) # screen dimensions in pixels
+ALIEN_ROWS = 3
+ALIENS_PER_ROW = 3
+HORIZONTAL_SPACING = WIDTH / ALIENS_PER_ROW # space between alien fleet
+                                    # columns
+VERTICAL_SPACING = HEIGHT / SCREEN_ROWS # space between alien fleet rows
+
+ALIEN_SPEED = 15
+ALIEN_MOVE_PERIOD = 5 # How many refresh cycles it takes for aliens to be
+                # redrawn
+
+PLAYER_SPEED = 15
+LASER_SPEED = 15
+
+RIGHT = (PLAYER_SPEED, 0) # increments of player movement
+LEFT = (-PLAYER_SPEED, 0) # ...
+DOWN = (0, ALIEN_SPEED) # increment of alien movement
+UP = (0, -LASER_SPEED)
+
+BLACK = (0, 0, 0)
+alien_velocity = [ALIEN_SPEED, 0] # first index is x velocity,
```

Comment:
Dont really see why this constant is needed if you have ALIEN_SPEED. Can and should have the 0 assigned to Y_VELOCITY or w/e you want to call it. Also, why not all caps for this?

Also, this is a global that gets modified? Can we instead have this as an attribute of Alien?